rotkehlchen/chain/ethereum/adex/adex.py

```diff
@@ -1037,38 +1011,49 @@ def get_events_history(
             to_timestamp: Timestamp,
     ) -> Dict[ChecksumAddress, ADXStakingHistory]:
         """Get the staking history events of the addresses in the AdEx protocol.
+
+        May raise RemoteError when there is a problem querying the Tom fee rewards
+        API or requesting the AdEx subgraph.
         """
         if self.graph is None:  # could not initialize graph
             return {}
 
         if reset_db_data is True:
             self.database.delete_adex_events_data()
 
-        fee_rewards = self._get_tom_pool_fee_rewards_from_api()
+        try:
+            fee_rewards = self._get_tom_pool_fee_rewards_from_api()
+            channel_ids_token = self._get_channel_ids_token(fee_rewards)
+            tom_pool_incentive = self._get_tom_pool_incentive(fee_rewards)
+        except (DeserializationError, RemoteError) as e:
+            self.msg_aggregator.add_error(
```

Comment:
Why not just let it go up? You are simply reraising it here and also logging a user message error. The decision whether to actually write a user message or not should lie with the caller of  `get_events_history`.

With this change I suggest you should also write that it may raise DeserializationError in its docstring and handleit at callers.

Reply:
I didn't let it go up as generally I haven't seen a DeserializationError bubbling up, I'd say we tend to re-raise them as RemoteError (I can be wrong), which is broadly captured.

I'll apply your suggestion here, on `get_events_history`, but what about on `get_balances`? Query all balances only handles `RemoteError` and `EthSyncError`. I re-raise it as a RemoteError for that reason. Do you want me to let it as it is (below) or also let it bubble up and handle DeserializationError on the caller?

```python
        try:
            fee_rewards = self._get_tom_pool_fee_rewards_from_api()
            channel_ids_token = self._get_channel_ids_token(fee_rewards)
        except (DeserializationError, RemoteError) as e:
            self.msg_aggregator.add_error(
                f'Got remote error querying AdEx balances due to: {str(e)}',
            )
            raise RemoteError(e) from e
```